iotilegateway/iotilegateway/supervisor/client.py

```diff
@@ -503,18 +503,21 @@ def post_info(self, name, message):
         else:
             try:
                 response = self._rpc_dispatcher.call_rpc(rpc_id, args)
-                if inspect.iscoroutine(response):
+                if inspect.isawaitable(response):
```

Comment:
I think I get why this is the better method to use over iscoroutine, but for our purposes does the distinction actually have the possibility of cropping up? Like, I don't see someone implementing `call_rpc` with a generator-based coroutine, right?

Reply:
I'm not sure if we'll see it popping up or not but `isawaitable` is the correct check for what we are trying to discover.  In many cases, `isawaitable` will be identical to `iscoroutine`, which will often be identical to `iscoroutinefunction` called on the parent function but not always and what we actually care about is: `was I handed something that I need to await before I get a result`, which is what `isawaitable directly answers.

An Example where `isawaitable` will work but `iscoroutine` won't is:
- we are returned an `asyncio.Future` object because the RPC itself was dispatched to a background thread pool for long-running execution without blocking the event loop.  In this case we could have forced the RPC implementation to `await` that future, turning the whole function into a coroutine, or it could just return the future and `isawaitable` will tell us we should await it, even though the function that returned it was just a normal function ("pass this object to another thread and get a thing back immediately, return the thing)